### PR TITLE
scripts: add support for arch linux on install dependencies

### PR DESCRIPTION
**Pull request by ashangit**
**Description of your changes:**

- add support for arch linux on install dependencies
- also ensure that tmp build folder is well deleted even if script failed

**Which issue is resolved by this Pull Request:**
Resolves #

### CODE REVIEW

File: install-dependencies.sh [hunk 1]
```diff
@@ -76,8 +88,11 @@ make WHAT=cmd/kube-apiserver
...
 
-cd ..
-rm -rf tmp
-
-# KUBEBUILDER_ASSETS has to be a single directory
-cp /usr/bin/etcd $(go env GOPATH)/bin
+# Build etcd
+git clone https://github.com/etcd-io/etcd.git
+cd etcd
+# master doesn't issue appropriate log message to validate etcd startup ("serving insecure client requests on")
```
Comment (zimnx) [inline install-dependencies.sh:94]: Please remove this comment, it's not related to Scylla Operator.

File: install-dependencies.sh [hunk 2]
```diff
@@ -76,8 +88,11 @@ make WHAT=cmd/kube-apiserver
...
+git checkout v3.4.15
+go mod vendor
+./build
+cp ./bin/etcd $(go env GOPATH)/bin
```
Comment (zimnx) [inline install-dependencies.sh:98]: add empty line at the end of the file

Comment (zimnx) [inline install-dependencies.sh:98]: `$(go env GOPATH)/bin` is used in two places now, could you extract it to variable?

**Comment by zimnx** [comment]
Please squash commits, rebase on top of recent master and if you could, provide maintainers access to your PR branch [using this tutorial](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork).

**Comment by ashangit** [comment]
Commits have been squashed
The  `Allow edits and access to secrets by maintainers`  is already checked so you should be able update the PR
For my knowledge (to improve other PR) which kind of pattern are you using for this repository (https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-request-merges)?

**Review by zimnx — APPROVED** [review]

**Pull request merged**